js/src/elements/dismiss.ts

```diff
@@ -0,0 +1,34 @@
+import { notifyContentUpdated } from '../utilities/notify-content-updated';
+
+export function initDismiss() {
+	window.addEventListener(
+		'click',
+		({ target }: Event) => {
+			const dismissButton = target instanceof Element && target.closest('[data-dismiss]');
+
+			if (!dismissButton) {
+				return;
+			}
+
+			const dismissTarget = dismissButton.closest('[data-dismissable]');
+
+			if (!dismissTarget) {
+				return;
+			}
+
+			dismissElement(dismissTarget);
+		},
+		{ passive: true }
+	);
+}
+
+async function dismissElement(dismissTarget: Element) {
+	if (dismissTarget.getAttribute('data-dismissable') === 'disappearing') {
+		dismissTarget.classList.add('disappearing');
```

Comment:
A third confirmation that I think we need to think about dismiss a bit and how we're going to use it as a selector.

We've got a few now:
[data-dismissable]
.disappearing
.notification-dismiss
.banner-dismiss

Reply:
After thinking about it for a while I don't think it's necessary to provide two options of how the element would be removed - slowly or immediately. I'm going to update so the default would "fade out" and if any component needs a different treatment - it can be adjusted within that's component css.

`.notification-dismiss` and `.banner-dismiss` classes were created to help with positioning the closing button correctly. I can update the selector to be `.banner[data-dismissable]` instead of `.banner.banner-dismiss`